packages/flutter/test/material/autocomplete_test.dart

```diff
@@ -453,4 +453,46 @@ void main() {
     checkOptionHighlight('chameleon', null);
     checkOptionHighlight('elephant', highlightColor);
   });
+
+  testWidgets('keyboard navigation keeps the highlighted option scrolled into view', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Autocomplete<String>(
+            optionsBuilder: (TextEditingValue textEditingValue) {
+              return kOptions.where((String option) {
+                return option.contains(textEditingValue.text.toLowerCase());
+              });
+            },
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.byType(TextFormField));
+    await tester.enterText(find.byType(TextFormField), 'e');
+    await tester.pump();
+    expect(find.byType(ListView), findsOneWidget);
+    final ListView list = find.byType(ListView).evaluate().first.widget as ListView;
+    expect(list.semanticChildCount, 6);
+
+    // Highlighted item should be at the top
+    expect(tester.getTopLeft(find.text('chameleon')).dy, equals(64.0));
```

Comment:
I am not super familiar with this area of the codebase, how do we know this i actually highlighted? We know it is present, but I don't know how the state is verified.

Reply:
Oh, `checkOptionHighlight` might be it.

Reply:
Good point. I moved `checkOptionHighlight` out of the previous test and used it here as well to verify the highlighted state of the scrolled item. (And uncovered a bug with the test). Thx.